components/brave_wallet/browser/blockchain_registry.cc

```diff
@@ -116,6 +116,33 @@ void BlockchainRegistry::GetAllTokens(const std::string& chain_id,
   std::move(callback).Run(std::move(tokens_copy));
 }
 
+void BlockchainRegistry::GetEthTokenListMap(
+    const std::vector<std::string>& chain_ids,
+    GetEthTokenListMapCallback callback) {
+  // Create a copy of token_list_map with only the chain_ids we want
+  TokenListMap token_list_map_copy;
+  for (const auto& chain_id : chain_ids) {
+    const auto key = GetTokenListKey(mojom::CoinType::ETH, chain_id);
+    // Skip if the key is not in the map.
+    if (!token_list_map_.contains(key)) {
+      continue;
+    }
+
+    // Otherwise, clone the vector of tokens.
+    const auto& tokens = token_list_map_[key];
+    std::vector<brave_wallet::mojom::BlockchainTokenPtr> tokens_copy(
+        tokens.size());
+    std::transform(
+        tokens.begin(), tokens.end(), tokens_copy.begin(),
+        [](const brave_wallet::mojom::BlockchainTokenPtr& current_token)
+            -> brave_wallet::mojom::BlockchainTokenPtr {
+          return current_token.Clone();
+        });
+    token_list_map_copy[chain_id] = std::move(tokens_copy);
+  }
+  std::move(callback).Run(std::move(token_list_map_copy));
+}
+
```

Comment:
I added this function to retrieve token registries for all ETH chain IDs in one call in order to avoid making repeated calls to this to the blockchain registry (per address, or worse, per address per chain).

This is used in two places, (1) at the beginning, when enqueuing all the JsonRpcService::GetERC20TokenBalances calls (to provide the token contract addresses param), and (2) at the end, when merging the results from the barrier callback, discovered tokens (to provide an easy way to get the BlockchainToken, from the token contract address).

Note: unlike in other areas, the keys are the chain IDs (not the CoinType concatenated with the chain ID).